core/src/main/java/org/apache/iceberg/actions/BaseRewriteDataFilesFileGroupInfo.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.actions;
+
+import org.apache.iceberg.StructLike;
+
+public class BaseRewriteDataFilesFileGroupInfo implements RewriteDataFiles.FileGroupInfo {
```

Comment:
Do we need to override `equals` and `hashCode` as this is used in the result map?

Reply:
Yes,  we will need the override `equals` and `hashCode` because I see we will use this `info` as keys in a [HashMap](https://github.com/apache/iceberg/pull/2591/files#diff-c04827ed079aa6cdede691f7ead4beaf30adc50f05da401a605215cdcd6dea4cR274).

Reply:
Not sure what do o here for equals and hashcode, since we can't get those for a StructLike without a Struct and at this point we don't have them. I'd be fine with changing the interface to just be a list of Pairs instead of key/value (or some new result type which has file group embedded)

Like
```
FileGroupRewriteResult {
  info
  filesAdded
  filesRewritten
  }
  ```

Reply:
I assume this is no longer used in maps?